ios/KaMPStarteriOS/ViewController.swift

```diff
@@ -43,20 +42,22 @@ class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSour
     // MARK: BreedModel Closures
     
     private func viewUpdate(for summary: ItemDataSummary) {
-        log.d(withMessage: {"View updating with \(summary.allItems.count) breeds"})
+        logger?.d(withMessage: {"View updating with \(summary.allItems.count) breeds"})
         data = summary.allItems
         breedTableView.reloadData()
     }
     
     private func errorUpdate(for errorMessage: String) {
-        log.e(withMessage: {"Displaying error: \(errorMessage)"})
+        logger?.e(withMessage: {"Displaying error: \(errorMessage)"})
         let alertController = UIAlertController(title: "error", message: errorMessage, preferredStyle: .actionSheet)
         alertController.addAction(UIAlertAction(title: "Dismiss", style: .default))
         present(alertController, animated: true, completion: nil)
     }
     
-    // MARK: TableView
-        
+}
+
+// MARK: - UITableViewDataSource
+extension ViewController: UITableViewDataSource {
```

Comment:
Something about this extension style makes my Kotlin brain uncomfortable but I should get myself more used to it and that's the point of this refactor.

Also, separately: I stared at this for a while worried that it would apply to all viewcontrollers and prevent us from adding a different viewcontroller that was a tableview with a different cell type. And then I realized it was extending `ViewController` and not `UIViewController`. Should we rename `ViewController` to `BreedViewController` or something like that?

Reply:
Something like BreedsViewController is probably a good idea, especially to help clarify for the Kotlin devs. iOS devs are more used to seeing both ViewController and UIViewController, because when you start an iOS app project (with UIKit, I have to specify now), you're provided with `class ViewController: UIViewController {...}`. Didn't occur to me that it might cause confusion.